packages/x-internals/src/rafThrottle/rafThrottle.test.ts

```diff
@@ -0,0 +1,231 @@
+import { expect } from 'vitest';
+import { rafThrottle } from './rafThrottle';
+
+describe('rafThrottle', () => {
+  beforeEach(() => {
+    vi.useFakeTimers();
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+    vi.useRealTimers();
+  });
+
+  it('should throttle multiple calls to execute at most once per frame', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    // Make multiple rapid calls
+    throttled(1);
+    throttled(2);
+    throttled(3);
+
+    // Function should not be called yet
+    expect(fn).not.toHaveBeenCalled();
+
+    // Execute pending RAF callback
+    vi.advanceTimersToNextFrame();
+
+    // Function should be called once with the last arguments
+    expect(fn).toHaveBeenCalledTimes(1);
+    expect(fn).toHaveBeenCalledWith(3);
+  });
+
+  it('should preserve the last arguments when called multiple times', async () => {
```

Comment:
Isn't this test the same as the one before it? They're both asserting it's being called with the last arguments. 

I think we can merge them